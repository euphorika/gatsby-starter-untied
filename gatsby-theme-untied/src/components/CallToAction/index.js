import React from 'react'
import PropTypes from 'prop-types'
import styles from './styles.module.styl'
import { Link } from 'gatsby'

const CallToAction = ({ button }) => (
  <section className={styles.callToActionSection}>
    <button className={styles.button}>
      <Link to={button.link}>
        <span> {button.text} </span>
      </Link>
    </button>
  </section>
)

CallToAction.propTypes = {
  button: PropTypes.shape({
    text: PropTypes.string.isRequired,
    link: PropTypes.string.isRequired,
  }),
}

export default CallToAction
