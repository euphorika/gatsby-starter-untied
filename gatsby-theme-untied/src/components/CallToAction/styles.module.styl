@require '../../themes/standard/config/breakpoints'
@require '../../themes/standard/config/colors'

.call-to-action-section
  margin: 0

.button
  display: block
  border-radius: 4px
  background-color: primary-color
  text-align: center
  font-size: 14px
  font-weight: 600
  padding: 15px
  width: 87%
  margin-top: 20px
  margin-bottom: 20px
  margin-left: auto
  margin-right: auto
  border-radius: 8px

.button span
  display: inline-block
  position: relative
  transition: 0.5s
  color: secondary-color

.button span:after
  content: '\00bb'
  position: absolute
  opacity: 0
  top: 0
  right: -20px
  transition: 0.5s

.button:hover span
  padding-right: 25px

.button:hover span:after
  opacity: 1
  right: 0
