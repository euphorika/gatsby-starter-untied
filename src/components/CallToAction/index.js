import React from 'react'

import styles from './styles.module.styl'

const CallToAction = () => (
  <section className={styles.CallToActionSection}>Call To Action</section>
)

export default CallToAction
