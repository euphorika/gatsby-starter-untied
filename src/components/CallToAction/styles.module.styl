@require '../../themes/standard/config/breakpoints'
@require '../../themes/standard/config/colors'

.call-to-action-section
  margin 0
