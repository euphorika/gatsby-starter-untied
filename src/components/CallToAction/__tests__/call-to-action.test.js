import React from 'React'
import renderer from 'react-test-renderer'
import CallToAction from '../'

describe('CallToAction', () => {
  it('renders correctly', () => {
    const button = {
      text: 'Text',
      link: '/',
    }

    const callToAction = <CallToAction button={button} />

    const tree = renderer.create(callToAction).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
