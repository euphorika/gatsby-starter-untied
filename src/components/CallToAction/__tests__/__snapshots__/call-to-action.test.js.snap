// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CallToAction renders correctly 1`] = `
<section
  className="callToActionSection"
>
  <button
    className="button"
  >
    <Link
      to="/"
    >
      <span>
         
        Text
         
      </span>
    </Link>
  </button>
</section>
`;
